thali/thalicryptomanager.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+var crypto = require('crypto');
+var fs = require('fs');
+var path = require('path');
+
+var pkcs12FileName = '/pkcs12.pfx';
+var password = 'password';
+var certname = 'certname';
+var country = 'country';
+var organization = 'organization';
+var macName = 'SHA256';
+var hashSizeInBytes = 16;
+
+module.exports = {
+
+/**
+* Creates a PKCS12 content, saves it to a file, extracts the public
+* key and returns it's SHA256 hash value.
+*/
+  getPublicKeyHash: function (callback) {
+    // get the path where the PKCS12 file is saved
+    Mobile.GetDocumentsPath(function (err, fileLocation) {
+      if (err) {
+        console.error("GetDocumentsPath err: ", err);
+        callback(err);
+      } else {
+        var file = path.join(fileLocation, pkcs12FileName);
+        getPKCS12Content(file, function(err, pkcs12Content) {
+          if(err) {
+            console.error('failed to get pkcs12 content');
+            callback(err);
+          }
+          // extract publick key
+          var publicKey = crypto.pkcs12.extractPublicKey(password, pkcs12Content);
+          if (publicKey.length <= 0) {
+            console.error('failed to extract public key');
+            callback('failed to extract publicKey');
+          }
+          console.log('generating SHA256 hash value');
```

Comment:
Is this really necessary in the log? Please review your console.log statements and ask yourself "Will anyone benefit by adding all of this to the log?" The more stuff is in the log the harder it is to use for much of anything. We already have a work item to add some kind of log leveling but we aren't there yet.